packages/lisk-core/src/controller.js

```diff
@@ -1,40 +1,171 @@
-'use strict';
-
-const { ModuleFactory } = require('@lisk-core/utils');
-const Bus = require('./bus');
 const Promise = require('bluebird');
+const fs = require('fs-extra');
+const config = require('./helpers/config');
+const ModuleFactory = require('./factories/modules');
+const ComponentFactory = require('./factories/components');
+const EventEmitterChannel = require('./channels/event_emitter');
+const Schema = require('./helpers/schema');
+const Bus = require('./bus');
+
+const logger = ComponentFactory.create('logger', config.components.logger);
 
 module.exports = class Controller {
 	constructor(options) {
-		console.log('Initializing controller');
+		logger.info('Initializing controller');
+		this.config = options;
 		this.modules = {};
+		this.channel = null;
 
 		// Setting up bus
-		if(!this.bus) {
+		if (!this.bus) {
 			this.bus = new Bus(this, {
```

Comment:
after decoupling Bus from Controller we can move this initialisation to a different file, and require the Bus instance as a commonjs module.